agent/lib/kontena/workers/service_pod_worker.rb

```diff
@@ -435,10 +439,13 @@ def sync_state_to_master(service_pod, service_container, error = nil)
           debug "skip #{service_pod} state sync at #{service_pod.deploy_rev}: unchanged"
         else
           debug "sync #{service_pod} state at #{service_pod.deploy_rev}: #{state}"
-          rpc_client.async.request('/node_service_pods/set_state', [node.id, state])
+          rpc_client.request('/node_service_pods/set_state', [node.id, state])
```

Comment:
This change to drop the `async` isn't entirely necessary for this PR, but I think this is probably a good idea.

As commented below, this means that the `@prev_state` doesn't get updated if the RPC fails, so the agent could re-sync the state later... although that would require something to call `sync_state_to_master` again, which won't happen just because the `sync_state_to_master` failed...